src/libs/maintaining-std.md

```diff
@@ -0,0 +1,132 @@
+- Final Comment Period (FCP). Coordinated by [`rfcbot`] that gives relevant teams a chance to review RFCs and PRs.
+
+## If you’re ever unsure…
+
+Maintaining the standard library can feel like a daunting responsibility! Through the automated reviewer assignment you’ll find yourself dropped into a lot of new contexts.
+
+Ping the `@rust-lang/libs` team on GitHub anytime. We’re all here to help!
+
+If you don’t think you’re the best person to review a PR then use [`bors`] to assign it to somebody else.
+
+## Finding reviews waiting for your input
+
+Please remember to regularly check https://rfcbot.rs/. Click on any occurrence of your nickname to go to a page like https://rfcbot.rs/fcp/SimonSapin that only shows the reviews that are waiting for your input.
+
+## Reviewing PRs
+
+As a member of the Libs team you’ll find yourself assigned to PRs that need reviewing, and your input requested on issues in the Rust project.
+
+### When is a RFC needed?
+
+New unstable features don’t need a RFC before they can be merged. They do need one before they can be stabilized. New impls for already stable traits don't need a RFC. If a new unstable feature is large or has a lot of design space to explore then you might want to block merging it on a RFC.
+
+### Is there any `unsafe`?
+
+Unsafe code blocks in the standard library need a comment explaining why they're [ok](https://doc.rust-lang.org/nomicon). There's a `tidy` lint that checks this. The unsafe code also needs to actually be ok.
+
+The rules around what's sound and what's not can be subtle. See the [Unsafe Code Guidelines WG] for current thinking, and consider pinging `@rust-lang/libs` and/or somebody from the WG if you're in any doubt.
+
+### Is that `#[inline]` right?
+
+Inlining is a trade-off between potential execution speed, compile time and code size.
+
+You should add `#[inline]`:
+
+- To public, small, non-generic functions.
+
+You shouldn’t need `#[inline]`:
+
+- On methods that take a generic argument.
+- On methods on traits that don’t have a default implementation.
+- On `const` items.
+
+#### What about `#[inline(always)]`?
+
+You should just about never need `#[inline(always)]`. It may be beneficial for private helper methods that are used in a limited number of places or for trivial operators. A micro benchmark should justify the attribute.
+
+### Is there any potential breakage?
+
+Breaking changes should be avoided when possible. [RFC 1105] lays the foundations for what constitutes a breaking change. Breakage may be deemed acceptable or not based on its actual impact, which can be approximated with a `crater` run.
+
+For changes where the value is high and the impact is high too, there are strategies for minimizing the impact:
+
+- Using compiler lints to try phase out broken behavior.
+
+### Could this affect inference?
+
+New impls on public traits for public items may cause inference to break when there are generics involved.
```

Comment:
I don't think this is limited to generics, right? Can't adding a second impl of some trait always cause inference failures, regardless of whether any of the impls is generic?

Reply:
Ah, I was mostly thinking of the case here where we'll optimistically type things when there's only a single trait impl. Say we have something like:

```rust
impl From<&'_ str> for Arc<str> { .. }

let b = Arc::from("a");
```

then we add:

```diff
impl From<&'_ str> for Arc<str> { .. }
+ impl From<&'_ str> for Arc<String> { .. }
```

then

```rust
let b = Arc::from("a");
```

will no longer compile, because we're relying on inference to figure out the `T` in `Box<T>`. I think this sort of thing comes up every now and again with types like `Arc`, and in new `FromIterator` impls.

Can we think of any other examples of real inference breakage that could make this section more useful?

Reply:
Another case to consider is deref coersion: https://github.com/rust-lang/rust/issues/51916